### DISCUSSION TITLE
Page queries taking longer to run without field extension

### DISCUSSION THREAD

**Discussion by jabrks** [gatsbyjs/gatsby#28518] [Help]
I have a site where some of the source data contains image paths that Gatsby can't resolve to file nodes by default. In order to mitigate that, we use a field extension to resolve them at build time. 

The original requirement for the image paths to be written this way no longer exists, so I set out to fix the paths and remove the field extension. I was mostly expecting the build time to stay the same, however after making the change the site now takes around 5 times longer on average to build, with most of this extra time being spent running the page queries.

Is there any reason that using a field extension for this purpose would result in a significantly faster build time?

**Comment by KyleAMathews** [comment]
Hard to say without seeing a reproduction. Is there some way you could share a minimal example of this happening?

**Reply by jabrks** [reply]
@KyleAMathews Sure, here you go

https://github.com/jabrks/gatsby-field-extension
https://github.com/jabrks/gatsby-no-field-extension

Sorry for the delay, I've only just managed to find time to create an example. In this case the site with the field extension took 93 seconds to build locally, where as the site without the field extension took 398 seconds. 

(cc @vladar)
